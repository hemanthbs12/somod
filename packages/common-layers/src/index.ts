import { existsSync } from "fs";
import { dirname, join } from "path";

// eslint-disable-next-line @typescript-eslint/no-namespace
export namespace layerLibraries {
  export const base: string[] = [
    "@solib/json-validator",
    "@solib/common-types-schemas",
    "@solib/errors",
    "lodash",
    "tslib",
    "uuid"
  ];
  export const customResource: string[] = ["@solib/cfn-custom-resource"];
  export const httpWrapper: string[] = ["@solib/http-lambda"];
}

export const getLocation = async (): Promise<string> => {
  let location = __dirname;
  while (!existsSync(join(location, "package.json"))) {
    location = dirname(location);
  }
  return location;
};
