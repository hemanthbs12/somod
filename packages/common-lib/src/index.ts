export * from "./json-validator";
export * from "./schema";
export * from "./errors";
