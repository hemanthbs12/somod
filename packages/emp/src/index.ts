import { rootCommand } from "@sodaru-cli/base";
import buildCommand from "./commands/build";
import deployCommand from "./commands/deploy";
import initCommand from "./commands/init";
import serveCommand from "./commands/serve";

const program = rootCommand("emp", [
  initCommand,
  buildCommand,
  serveCommand,
  deployCommand
]);

export default program;
