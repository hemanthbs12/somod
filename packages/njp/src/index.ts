import { rootCommand } from "@solib/cli-base";
import buildCommand from "./commands/build";
import serveCommand from "./commands/serve";
import initCommand from "./commands/init";
import prepareCommand from "./commands/prepare";

const program = rootCommand("njp", [
  initCommand,
  buildCommand,
  prepareCommand,
  serveCommand
]);

export default program;
