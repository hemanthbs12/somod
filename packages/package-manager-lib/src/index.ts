export * from "./tasks/packageJson";

export * from "./tasks/ignoreFiles";

export * from "./tasks/tsConfigBuildJson";

export * from "./tasks/init";

export * from "./utils/constants";
