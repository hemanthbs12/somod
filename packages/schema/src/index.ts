export * from "./common/keywords";
export * from "./serverless-template/components/expression";
export * from "./serverless-template/components/keywords";
export * from "./serverless-template/resources/all";
export * from "./serverless-template/resources/custom";
export * from "./serverless-template/resources/function";
export * from "./serverless-template/resources/functionLayer";
export * from "./ui-config/index";
export { default as parameters } from "./parameters/index";
export { default as serverless } from "./serverless-template/index";
export { default as uiConfig } from "./ui-config/index";
