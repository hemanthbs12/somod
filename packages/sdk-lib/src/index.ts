export * from "./tasks/packageJson";

export * from "./tasks/ignoreFiles";

export * from "./tasks/tsConfigBuildJson";

export * from "./tasks/init";

export * from "./tasks/build";

export * from "./tasks/link";

export * from "./tasks/dev";

export * from "./tasks/serverless";

export * from "./utils/constants";
