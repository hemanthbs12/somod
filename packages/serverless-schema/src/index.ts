import Ajv, { AnySchemaObject, ValidateFunction } from "ajv";
import { readFile } from "fs/promises";
import { get as httpGet } from "http";
import { get as httpsGet } from "https";
import { join } from "path";
import { URL } from "url";
import { isPlainObject, isArray, isString } from "lodash";

const idBase = "https://json-schema.sodaru.com";

const loadFromFile = async (path: string): Promise<string> => {
  const content = readFile(path, { encoding: "utf8" });
  return content;
};

const loadFromWeb = (uri: string): Promise<string> => {
  const requestFunction = uri.startsWith("https://") ? httpsGet : httpGet;
  return new Promise((resolve, reject) => {
    const request = requestFunction(uri, res => {
      if (!(res.statusCode >= 200 && res.statusCode < 300)) {
        reject(
          new Error(
            `Invalid response from ${uri}, statusCode : ${res.statusCode}`
          )
        );
      }
      const chunks: string[] = [];
      res.on("data", chunk => {
        chunks.push(chunk);
      });
      res.on("end", () => {
        resolve(chunks.join(""));
      });
    });
    request.on("error", e => {
      reject(e);
    });
  });
};

const getModuleFromUri = (uri: string): string => {
  let moduleName = null;
  if (uri.startsWith(idBase)) {
    const url = new URL(uri);
    const paths = url.pathname.split("/");
    if (paths.length >= 2) {
      moduleName = paths[1].startsWith("@")
        ? paths[1] + "/" + paths[2]
        : paths[1];
    }
  }
  if (!moduleName) {
    throw new Error(`Could not find modulename from ${uri}`);
  }
  return moduleName;
};

const uriReferenceToUri = (input: string, base: string): string => {
  let _input = input;
  if (_input.indexOf("/node_modules/")) {
    const baseModuleName = getModuleFromUri(base) as string;
    const replacement = baseModuleName.startsWith(`@`) ? "/../../" : "/../";
    _input = _input.replace("/node_modules/", replacement);
  }
  const url = new URL(_input, base);
  return url.toString();
};

const extractReferences = (
  schema: string
): { schema: AnySchemaObject; $refs: string[] } => {
  const correctedSchema = schema.replaceAll(
    "https://json-schema.org/draft-07/schema", // vs-code extension redhat.vscode-yaml requires https://json-schema.org/draft-07/schema
    "http://json-schema.org/draft-07/schema" // ajv requires http://json-schema.org/draft-07/schema
  );

  const schemaObj: AnySchemaObject = JSON.parse(correctedSchema);
  schemaObj.$schema = uriReferenceToUri(schemaObj.$schema, schemaObj.$id);

  const $_refs: string[] = [];
  const objectsToBeParsed = [schemaObj];
  while (objectsToBeParsed.length > 0) {
    const objectToBeParsed = objectsToBeParsed.shift();
    if (isPlainObject(objectToBeParsed)) {
      if (isString(objectToBeParsed.$ref)) {
        const $_ref = uriReferenceToUri(objectToBeParsed.$ref, schemaObj.$id);
        objectToBeParsed.$ref = $_ref;
        $_refs.push($_ref);
      }
      objectsToBeParsed.push(...Object.values(objectToBeParsed));
    } else if (isArray(objectToBeParsed)) {
      objectsToBeParsed.push(...objectToBeParsed);
    }
  }
  const $refs = $_refs
    .map($ref => {
      const url = new URL($ref);
      return `${url.protocol}//${url.host}${url.pathname}`; // drop fragment in uri
    })
    .filter($ref => $ref != schemaObj.$id); // filter references to same file

  return { schema: schemaObj, $refs };
};

const visitedIds: string[] = [];

const resetVisitedIds = () => {
  // RESET visited Ids
  while (visitedIds.length > 0) {
    visitedIds.pop();
  }
  visitedIds.push("http://json-schema.org/draft-07/schema");
};

const parseAndLoad = async (
  id: string,
  ajv: Ajv,
  dir: string,
  rootModuleName: string,
  loadType: "addSchema" | "addMetaSchema" | "compile"
): Promise<void | ValidateFunction> => {
  if (!visitedIds.includes(id)) {
    visitedIds.push(id);

    let fileContent: string = null;
    if (id.startsWith(idBase)) {
      const moduleName = getModuleFromUri(id);
      const moduleRootDir =
        moduleName == rootModuleName
          ? dir
          : join(dir, "node_modules", moduleName);
      const relativePath = id.substr(`${idBase}/${moduleName}/`.length);
      fileContent = await loadFromFile(join(moduleRootDir, relativePath));
    } else {
      fileContent = await loadFromWeb(id);
    }

    const { schema, $refs } = extractReferences(fileContent);
    await parseAndLoad(
      schema.$schema,
      ajv,
      dir,
      rootModuleName,
      "addMetaSchema"
    );
    await Promise.all(
      $refs.map($ref =>
        parseAndLoad($ref, ajv, dir, rootModuleName, "addSchema")
      )
    );

    if (loadType == "addMetaSchema") {
      if (!ajv.getSchema(id)) {
        ajv.addMetaSchema(schema);
      }
    } else if (loadType == "addSchema") {
      if (!ajv.getSchema(id)) {
        ajv.addSchema(schema);
      }
    } else {
      return ajv.compile(schema);
    }
  }
};

export const compile = async (
  dir: string,
  ajv: Ajv
): Promise<ValidateFunction> => {
  const packageJsonStr = await readFile(join(dir, "package.json"), {
    encoding: "utf8"
  });
  const packageJson = JSON.parse(packageJsonStr);
  const id = packageJson.serverlessSchema;

  const thisModuleName = "@somod/serverless-schema";

  const schemaToCompile =
    id || `${idBase}/${thisModuleName}/schemas/index.json`;

  resetVisitedIds();

  const validateFunction = (await parseAndLoad(
    schemaToCompile,
    ajv,
    dir,
    packageJson.name,
    "compile"
  )) as ValidateFunction;

  return validateFunction;
};
