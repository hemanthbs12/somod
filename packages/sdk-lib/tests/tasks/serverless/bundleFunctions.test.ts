import { childProcess } from "@solib/cli-base";
import { createFiles, createTempDir, deleteDir } from "@sodev/test-utils";
import { readFile } from "fs/promises";
import { join } from "path";
import { bundleFunctions } from "../../../src";

describe("Test Task bundleFunctions", () => {
  let dir: string;
  beforeEach(() => {
    dir = createTempDir();
  });

  afterEach(() => {
    deleteDir(dir);
  });

  test("with no functions", async () => {
    await expect(bundleFunctions(dir)).resolves.toBeUndefined();
  });

  test("with empty functions dir", async () => {
    createFiles(dir, { "serverless/functions/": "" });
    await expect(bundleFunctions(dir)).resolves.toBeUndefined();
  });

  test("with one function and no exclude file", async () => {
    createFiles(dir, { "serverless/functions/f1.ts": "" });
    await expect(bundleFunctions(dir)).rejects.toHaveProperty(
      "message",
      `ENOENT: no such file or directory, open '${join(
        dir,
        "build/serverless/functions/f1/exclude.json"
      )}'`
    );
  });

  test("with one function and valid exclude file", async () => {
    createFiles(dir, {
      "build/serverless/functions/f1/exclude.json": '{"external":["lodash"]}',
      "serverless/functions/f1.ts":
        'import {difference} from "lodash"; export const diff = (a:string[], b:string[]):string[] => {return difference(a,b);}'
    });
    await expect(bundleFunctions(dir)).resolves.toBeUndefined();
    await expect(
      readFile(join(dir, "build/serverless/functions/f1/index.js"), {
        encoding: "utf8"
      })
    ).resolves.toMatchSnapshot();
  });

  test("with multiple functions", async () => {
    createFiles(dir, {
      "build/serverless/functions/f1/exclude.json": '{"external":["lodash"]}',
      "build/serverless/functions/f2/exclude.json": '{"external":["aws-sdk"]}',
      "serverless/functions/f1.ts":
        'import {difference} from "lodash"; export const diff = (a:string[], b:string[]):string[] => {return difference(a,b);}',
      "serverless/functions/f2.ts":
        'import {DynamoDB} from "aws-sdk";import difference from "lodash/difference"; export const conn = new DynamoDB(); export const diff = (a:string[], b:string[]):string[] => {return difference(a,b);}',
      "package.json": JSON.stringify(
        {
          name: "sample",
          version: "1.0.0",
          dependencies: { lodash: "^4.17.21" }
        },
        null,
        2
      )
    });

    await childProcess(dir, process.platform == "win32" ? "npm.cmd" : "npm", [
      "install"
    ]);

    await expect(bundleFunctions(dir)).resolves.toBeUndefined();

    await expect(
      readFile(join(dir, "build/serverless/functions/f1/index.js"), {
        encoding: "utf8"
      })
    ).resolves.toMatchSnapshot();
    await expect(
      readFile(join(dir, "build/serverless/functions/f2/index.js"), {
        encoding: "utf8"
      })
    ).resolves.toMatchSnapshot();
  });
});
