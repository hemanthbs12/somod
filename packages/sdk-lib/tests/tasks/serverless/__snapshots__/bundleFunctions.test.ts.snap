// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Task bundleFunctions with multiple functions 1`] = `
"var e=Object.defineProperty;var s=Object.getOwnPropertyDescriptor;var g=Object.getOwnPropertyNames;var c=Object.prototype.hasOwnProperty;var d=t=>e(t,\\"__esModule\\",{value:!0});var m=(t,r)=>{for(var i in r)e(t,i,{get:r[i],enumerable:!0})},p=(t,r,i,f)=>{if(r&&typeof r==\\"object\\"||typeof r==\\"function\\")for(let n of g(r))!c.call(t,n)&&(i||n!==\\"default\\")&&e(t,n,{get:()=>r[n],enumerable:!(f=s(r,n))||f.enumerable});return t};var u=(t=>(r,i)=>t&&t.get(r)||(i=p(d({}),r,1),t&&t.set(r,i),i))(typeof WeakMap!=\\"undefined\\"?new WeakMap:0);var a={};m(a,{diff:()=>x});var o=require(\\"lodash\\"),x=(t,r)=>(0,o.difference)(t,r);module.exports=u(a);0&&(module.exports={diff});
"
`;

exports[`Test Task bundleFunctions with multiple functions 2`] = `
"var qt=Object.create;var b=Object.defineProperty;var yt=Object.getOwnPropertyDescriptor;var Ot=Object.getOwnPropertyNames;var Ct=Object.getPrototypeOf,St=Object.prototype.hasOwnProperty;var A=e=>b(e,\\"__esModule\\",{value:!0});var n=(e,r)=>()=>(r||e((r={exports:{}}).exports,r),r.exports),jt=(e,r)=>{for(var t in r)b(e,t,{get:r[t],enumerable:!0})},D=(e,r,t,i)=>{if(r&&typeof r==\\"object\\"||typeof r==\\"function\\")for(let a of Ot(r))!St.call(e,a)&&(t||a!==\\"default\\")&&b(e,a,{get:()=>r[a],enumerable:!(i=yt(r,a))||i.enumerable});return e},mt=(e,r)=>D(A(b(e!=null?qt(Ct(e)):{},\\"default\\",!r&&e&&e.__esModule?{get:()=>e.default,enumerable:!0}:{value:e,enumerable:!0})),e),wt=(e=>(r,t)=>e&&e.get(r)||(t=D(A({}),r,1),e&&e.set(r,t),t))(typeof WeakMap!=\\"undefined\\"?new WeakMap:0);var H=n((to,N)=>{var Tt=typeof global==\\"object\\"&&global&&global.Object===Object&&global;N.exports=Tt});var q=n((no,F)=>{var It=H(),Pt=typeof self==\\"object\\"&&self&&self.Object===Object&&self,At=It||Pt||Function(\\"return this\\")();F.exports=At});var y=n((io,M)=>{var Dt=q(),Nt=Dt.Symbol;M.exports=Nt});var z=n((ao,L)=>{var E=y(),G=Object.prototype,Ht=G.hasOwnProperty,Ft=G.toString,f=E?E.toStringTag:void 0;function Mt(e){var r=Ht.call(e,f),t=e[f];try{e[f]=void 0;var i=!0}catch{}var a=Ft.call(e);return i&&(r?e[f]=t:delete e[f]),a}L.exports=Mt});var U=n((oo,R)=>{var Et=Object.prototype,Gt=Et.toString;function Lt(e){return Gt.call(e)}R.exports=Lt});var j=n((so,k)=>{var $=y(),zt=z(),Rt=U(),Ut=\\"[object Null]\\",$t=\\"[object Undefined]\\",K=$?$.toStringTag:void 0;function Kt(e){return e==null?e===void 0?$t:Ut:K&&K in Object(e)?zt(e):Rt(e)}k.exports=Kt});var m=n((uo,J)=>{function kt(e){var r=typeof e;return e!=null&&(r==\\"object\\"||r==\\"function\\")}J.exports=kt});var w=n((co,V)=>{var Jt=j(),Vt=m(),Wt=\\"[object AsyncFunction]\\",Bt=\\"[object Function]\\",Xt=\\"[object GeneratorFunction]\\",Yt=\\"[object Proxy]\\";function Zt(e){if(!Vt(e))return!1;var r=Jt(e);return r==Bt||r==Xt||r==Wt||r==Yt}V.exports=Zt});var B=n((po,W)=>{var Qt=q(),en=Qt[\\"__core-js_shared__\\"];W.exports=en});var Z=n((ho,Y)=>{var T=B(),X=function(){var e=/[^.]+$/.exec(T&&T.keys&&T.keys.IE_PROTO||\\"\\");return e?\\"Symbol(src)_1.\\"+e:\\"\\"}();function rn(e){return!!X&&X in e}Y.exports=rn});var ee=n((lo,Q)=>{var tn=Function.prototype,nn=tn.toString;function an(e){if(e!=null){try{return nn.call(e)}catch{}try{return e+\\"\\"}catch{}}return\\"\\"}Q.exports=an});var te=n((fo,re)=>{var on=w(),sn=Z(),un=m(),cn=ee(),pn=/[\\\\\\\\^$.*+?()[\\\\]{}|]/g,hn=/^\\\\[object .+?Constructor\\\\]$/,ln=Function.prototype,fn=Object.prototype,dn=ln.toString,gn=fn.hasOwnProperty,vn=RegExp(\\"^\\"+dn.call(gn).replace(pn,\\"\\\\\\\\$&\\").replace(/hasOwnProperty|(function).*?(?=\\\\\\\\\\\\()| for .+?(?=\\\\\\\\\\\\])/g,\\"$1.*?\\")+\\"$\\");function xn(e){if(!un(e)||sn(e))return!1;var r=on(e)?vn:hn;return r.test(cn(e))}re.exports=xn});var ie=n((go,ne)=>{function _n(e,r){return e==null?void 0:e[r]}ne.exports=_n});var O=n((vo,ae)=>{var bn=te(),qn=ie();function yn(e,r){var t=qn(e,r);return bn(t)?t:void 0}ae.exports=yn});var d=n((xo,oe)=>{var On=O(),Cn=On(Object,\\"create\\");oe.exports=Cn});var ce=n((_o,ue)=>{var se=d();function Sn(){this.__data__=se?se(null):{},this.size=0}ue.exports=Sn});var he=n((bo,pe)=>{function jn(e){var r=this.has(e)&&delete this.__data__[e];return this.size-=r?1:0,r}pe.exports=jn});var fe=n((qo,le)=>{var mn=d(),wn=\\"__lodash_hash_undefined__\\",Tn=Object.prototype,In=Tn.hasOwnProperty;function Pn(e){var r=this.__data__;if(mn){var t=r[e];return t===wn?void 0:t}return In.call(r,e)?r[e]:void 0}le.exports=Pn});var ge=n((yo,de)=>{var An=d(),Dn=Object.prototype,Nn=Dn.hasOwnProperty;function Hn(e){var r=this.__data__;return An?r[e]!==void 0:Nn.call(r,e)}de.exports=Hn});var xe=n((Oo,ve)=>{var Fn=d(),Mn=\\"__lodash_hash_undefined__\\";function En(e,r){var t=this.__data__;return this.size+=this.has(e)?0:1,t[e]=Fn&&r===void 0?Mn:r,this}ve.exports=En});var be=n((Co,_e)=>{var Gn=ce(),Ln=he(),zn=fe(),Rn=ge(),Un=xe();function p(e){var r=-1,t=e==null?0:e.length;for(this.clear();++r<t;){var i=e[r];this.set(i[0],i[1])}}p.prototype.clear=Gn;p.prototype.delete=Ln;p.prototype.get=zn;p.prototype.has=Rn;p.prototype.set=Un;_e.exports=p});var ye=n((So,qe)=>{function $n(){this.__data__=[],this.size=0}qe.exports=$n});var Ce=n((jo,Oe)=>{function Kn(e,r){return e===r||e!==e&&r!==r}Oe.exports=Kn});var g=n((mo,Se)=>{var kn=Ce();function Jn(e,r){for(var t=e.length;t--;)if(kn(e[t][0],r))return t;return-1}Se.exports=Jn});var me=n((wo,je)=>{var Vn=g(),Wn=Array.prototype,Bn=Wn.splice;function Xn(e){var r=this.__data__,t=Vn(r,e);if(t<0)return!1;var i=r.length-1;return t==i?r.pop():Bn.call(r,t,1),--this.size,!0}je.exports=Xn});var Te=n((To,we)=>{var Yn=g();function Zn(e){var r=this.__data__,t=Yn(r,e);return t<0?void 0:r[t][1]}we.exports=Zn});var Pe=n((Io,Ie)=>{var Qn=g();function ei(e){return Qn(this.__data__,e)>-1}Ie.exports=ei});var De=n((Po,Ae)=>{var ri=g();function ti(e,r){var t=this.__data__,i=ri(t,e);return i<0?(++this.size,t.push([e,r])):t[i][1]=r,this}Ae.exports=ti});var He=n((Ao,Ne)=>{var ni=ye(),ii=me(),ai=Te(),oi=Pe(),si=De();function h(e){var r=-1,t=e==null?0:e.length;for(this.clear();++r<t;){var i=e[r];this.set(i[0],i[1])}}h.prototype.clear=ni;h.prototype.delete=ii;h.prototype.get=ai;h.prototype.has=oi;h.prototype.set=si;Ne.exports=h});var Me=n((Do,Fe)=>{var ui=O(),ci=q(),pi=ui(ci,\\"Map\\");Fe.exports=pi});var Le=n((No,Ge)=>{var Ee=be(),hi=He(),li=Me();function fi(){this.size=0,this.__data__={hash:new Ee,map:new(li||hi),string:new Ee}}Ge.exports=fi});var Re=n((Ho,ze)=>{function di(e){var r=typeof e;return r==\\"string\\"||r==\\"number\\"||r==\\"symbol\\"||r==\\"boolean\\"?e!==\\"__proto__\\":e===null}ze.exports=di});var v=n((Fo,Ue)=>{var gi=Re();function vi(e,r){var t=e.__data__;return gi(r)?t[typeof r==\\"string\\"?\\"string\\":\\"hash\\"]:t.map}Ue.exports=vi});var Ke=n((Mo,$e)=>{var xi=v();function _i(e){var r=xi(this,e).delete(e);return this.size-=r?1:0,r}$e.exports=_i});var Je=n((Eo,ke)=>{var bi=v();function qi(e){return bi(this,e).get(e)}ke.exports=qi});var We=n((Go,Ve)=>{var yi=v();function Oi(e){return yi(this,e).has(e)}Ve.exports=Oi});var Xe=n((Lo,Be)=>{var Ci=v();function Si(e,r){var t=Ci(this,e),i=t.size;return t.set(e,r),this.size+=t.size==i?0:1,this}Be.exports=Si});var Ze=n((zo,Ye)=>{var ji=Le(),mi=Ke(),wi=Je(),Ti=We(),Ii=Xe();function l(e){var r=-1,t=e==null?0:e.length;for(this.clear();++r<t;){var i=e[r];this.set(i[0],i[1])}}l.prototype.clear=ji;l.prototype.delete=mi;l.prototype.get=wi;l.prototype.has=Ti;l.prototype.set=Ii;Ye.exports=l});var er=n((Ro,Qe)=>{var Pi=\\"__lodash_hash_undefined__\\";function Ai(e){return this.__data__.set(e,Pi),this}Qe.exports=Ai});var tr=n((Uo,rr)=>{function Di(e){return this.__data__.has(e)}rr.exports=Di});var ir=n(($o,nr)=>{var Ni=Ze(),Hi=er(),Fi=tr();function C(e){var r=-1,t=e==null?0:e.length;for(this.__data__=new Ni;++r<t;)this.add(e[r])}C.prototype.add=C.prototype.push=Hi;C.prototype.has=Fi;nr.exports=C});var or=n((Ko,ar)=>{function Mi(e,r,t,i){for(var a=e.length,o=t+(i?1:-1);i?o--:++o<a;)if(r(e[o],o,e))return o;return-1}ar.exports=Mi});var ur=n((ko,sr)=>{function Ei(e){return e!==e}sr.exports=Ei});var pr=n((Jo,cr)=>{function Gi(e,r,t){for(var i=t-1,a=e.length;++i<a;)if(e[i]===r)return i;return-1}cr.exports=Gi});var lr=n((Vo,hr)=>{var Li=or(),zi=ur(),Ri=pr();function Ui(e,r,t){return r===r?Ri(e,r,t):Li(e,zi,t)}hr.exports=Ui});var dr=n((Wo,fr)=>{var $i=lr();function Ki(e,r){var t=e==null?0:e.length;return!!t&&$i(e,r,0)>-1}fr.exports=Ki});var vr=n((Bo,gr)=>{function ki(e,r,t){for(var i=-1,a=e==null?0:e.length;++i<a;)if(t(r,e[i]))return!0;return!1}gr.exports=ki});var _r=n((Xo,xr)=>{function Ji(e,r){for(var t=-1,i=e==null?0:e.length,a=Array(i);++t<i;)a[t]=r(e[t],t,e);return a}xr.exports=Ji});var qr=n((Yo,br)=>{function Vi(e){return function(r){return e(r)}}br.exports=Vi});var Or=n((Zo,yr)=>{function Wi(e,r){return e.has(r)}yr.exports=Wi});var Sr=n((Qo,Cr)=>{var Bi=ir(),Xi=dr(),Yi=vr(),Zi=_r(),Qi=qr(),ea=Or(),ra=200;function ta(e,r,t,i){var a=-1,o=Xi,u=!0,s=e.length,x=[],bt=r.length;if(!s)return x;t&&(r=Zi(r,Qi(t))),i?(o=Yi,u=!1):r.length>=ra&&(o=ea,u=!1,r=new Bi(r));e:for(;++a<s;){var c=e[a],_=t==null?c:t(c);if(c=i||c!==0?c:0,u&&_===_){for(var P=bt;P--;)if(r[P]===_)continue e;x.push(c)}else o(r,_,i)||x.push(c)}return x}Cr.exports=ta});var mr=n((es,jr)=>{function na(e,r){for(var t=-1,i=r.length,a=e.length;++t<i;)e[a+t]=r[t];return e}jr.exports=na});var S=n((rs,wr)=>{function ia(e){return e!=null&&typeof e==\\"object\\"}wr.exports=ia});var Ir=n((ts,Tr)=>{var aa=j(),oa=S(),sa=\\"[object Arguments]\\";function ua(e){return oa(e)&&aa(e)==sa}Tr.exports=ua});var Nr=n((ns,Dr)=>{var Pr=Ir(),ca=S(),Ar=Object.prototype,pa=Ar.hasOwnProperty,ha=Ar.propertyIsEnumerable,la=Pr(function(){return arguments}())?Pr:function(e){return ca(e)&&pa.call(e,\\"callee\\")&&!ha.call(e,\\"callee\\")};Dr.exports=la});var Fr=n((is,Hr)=>{var fa=Array.isArray;Hr.exports=fa});var Lr=n((as,Gr)=>{var Mr=y(),da=Nr(),ga=Fr(),Er=Mr?Mr.isConcatSpreadable:void 0;function va(e){return ga(e)||da(e)||!!(Er&&e&&e[Er])}Gr.exports=va});var Ur=n((os,Rr)=>{var xa=mr(),_a=Lr();function zr(e,r,t,i,a){var o=-1,u=e.length;for(t||(t=_a),a||(a=[]);++o<u;){var s=e[o];r>0&&t(s)?r>1?zr(s,r-1,t,i,a):xa(a,s):i||(a[a.length]=s)}return a}Rr.exports=zr});var I=n((ss,$r)=>{function ba(e){return e}$r.exports=ba});var kr=n((us,Kr)=>{function qa(e,r,t){switch(t.length){case 0:return e.call(r);case 1:return e.call(r,t[0]);case 2:return e.call(r,t[0],t[1]);case 3:return e.call(r,t[0],t[1],t[2])}return e.apply(r,t)}Kr.exports=qa});var Wr=n((cs,Vr)=>{var ya=kr(),Jr=Math.max;function Oa(e,r,t){return r=Jr(r===void 0?e.length-1:r,0),function(){for(var i=arguments,a=-1,o=Jr(i.length-r,0),u=Array(o);++a<o;)u[a]=i[r+a];a=-1;for(var s=Array(r+1);++a<r;)s[a]=i[a];return s[r]=t(u),ya(e,this,s)}}Vr.exports=Oa});var Xr=n((ps,Br)=>{function Ca(e){return function(){return e}}Br.exports=Ca});var Zr=n((hs,Yr)=>{var Sa=O(),ja=function(){try{var e=Sa(Object,\\"defineProperty\\");return e({},\\"\\",{}),e}catch{}}();Yr.exports=ja});var rt=n((ls,et)=>{var ma=Xr(),Qr=Zr(),wa=I(),Ta=Qr?function(e,r){return Qr(e,\\"toString\\",{configurable:!0,enumerable:!1,value:ma(r),writable:!0})}:wa;et.exports=Ta});var nt=n((fs,tt)=>{var Ia=800,Pa=16,Aa=Date.now;function Da(e){var r=0,t=0;return function(){var i=Aa(),a=Pa-(i-t);if(t=i,a>0){if(++r>=Ia)return arguments[0]}else r=0;return e.apply(void 0,arguments)}}tt.exports=Da});var at=n((ds,it)=>{var Na=rt(),Ha=nt(),Fa=Ha(Na);it.exports=Fa});var st=n((gs,ot)=>{var Ma=I(),Ea=Wr(),Ga=at();function La(e,r){return Ga(Ea(e,r,Ma),e+\\"\\")}ot.exports=La});var ct=n((vs,ut)=>{var za=9007199254740991;function Ra(e){return typeof e==\\"number\\"&&e>-1&&e%1==0&&e<=za}ut.exports=Ra});var ht=n((xs,pt)=>{var Ua=w(),$a=ct();function Ka(e){return e!=null&&$a(e.length)&&!Ua(e)}pt.exports=Ka});var ft=n((_s,lt)=>{var ka=ht(),Ja=S();function Va(e){return Ja(e)&&ka(e)}lt.exports=Va});var vt=n((bs,gt)=>{var Wa=Sr(),Ba=Ur(),Xa=st(),dt=ft(),Ya=Xa(function(e,r){return dt(e)?Wa(e,Ba(r,1,dt,!0)):[]});gt.exports=Ya});var eo={};jt(eo,{conn:()=>Za,diff:()=>Qa});var xt=require(\\"aws-sdk\\"),_t=mt(vt()),Za=new xt.DynamoDB,Qa=(e,r)=>(0,_t.default)(e,r);module.exports=wt(eo);0&&(module.exports={conn,diff});
"
`;

exports[`Test Task bundleFunctions with one function and valid exclude file 1`] = `
"var e=Object.defineProperty;var s=Object.getOwnPropertyDescriptor;var g=Object.getOwnPropertyNames;var c=Object.prototype.hasOwnProperty;var d=t=>e(t,\\"__esModule\\",{value:!0});var m=(t,r)=>{for(var i in r)e(t,i,{get:r[i],enumerable:!0})},p=(t,r,i,f)=>{if(r&&typeof r==\\"object\\"||typeof r==\\"function\\")for(let n of g(r))!c.call(t,n)&&(i||n!==\\"default\\")&&e(t,n,{get:()=>r[n],enumerable:!(f=s(r,n))||f.enumerable});return t};var u=(t=>(r,i)=>t&&t.get(r)||(i=p(d({}),r,1),t&&t.set(r,i),i))(typeof WeakMap!=\\"undefined\\"?new WeakMap:0);var a={};m(a,{diff:()=>x});var o=require(\\"lodash\\"),x=(t,r)=>(0,o.difference)(t,r);module.exports=u(a);0&&(module.exports={diff});
"
`;
